File: pe006/pe006-naive.go
/*
Project Euler, Problem 6:

Find the difference between the sum of the squares of the first one hundred
natural numbers and the square of the sum.

https://projecteuler.net/problem=6
*/

package main

import "fmt"

const max = 100

func main() {
	sum, squares := 0, 0
	for i := 1; i <= max; i++ {
		sum += i
		squares += i * i
	}
	fmt.Println(sum*sum - squares)
}

File: pe006/pe006-naive.py
#!/usr/bin/python

"""
Project Euler, Problem 6:

Find the difference between the sum of the squares of the first one hundred
natural numbers and the square of the sum.

https://projecteuler.net/problem=6
"""

max_number = 100

def main():
    the_sum = sum(list(range(1, max_number + 1)))
    squares = sum([ i * i for i in range(1, max_number + 1) ])
    print(the_sum * the_sum - squares)

if __name__ == "__main__":
    main()
